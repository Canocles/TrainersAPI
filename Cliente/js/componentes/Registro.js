import React from 'react'

class Registro extends React.Component {
    render() {
        return (
            <div className="row" onSubmit={this.registro}>
                <div className="col-md-4 col-md-offset-4">
                    <form>
                        <h1>Únete a nuestra comunidad!</h1>
                        <div className="form-group">
                            <label className="control-label">Email</label>
                            <input ref={campo => this.email = campo} type="email" name="email" placeholder="Introduzca su email" className="form-control" required/>
                        </div>
                        <div className="form-group">
                            <label className="control-label">Usuario</label>
                            <input ref={campo => this.login = campo} type="text" name="login" placeholder="Introduzca su nombre de usuario" className="form-control" required/>
                        </div>
                        <div className="form-group">
                            <label className="control-label">Password</label>
                            <input ref={campo => this.password = campo} type="password" name="password" placeholder="Introduzca su contraseña" className="form-control" required/>
                        </div>
                        <div className="form-group">
                            <a className="btn btn-primary btn-lg" onClick={() => { this.props.handleregistro(this.email.value, this.login.value, this.password.value) }}>
                                Regístrate
                            </a>
                        </div>
                    </form>
                </div>
            </div>
        )    
    }
}

export default Registro