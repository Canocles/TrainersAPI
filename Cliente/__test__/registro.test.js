import React from 'react';
import renderer from 'react-test-renderer'
import Registro from '../src/componentes/Registro';

describe('Registro (snpashot)', () => {
  it('El componente de Registro se renderiza correctamente', () => {
    const component = renderer.create(<Registro />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
