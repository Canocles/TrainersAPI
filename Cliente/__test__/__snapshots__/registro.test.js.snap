// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registro (snpashot) El componente de Registro se renderiza correctamente 1`] = `
<div
  className="row"
>
  <div
    className="col-md-4 col-md-offset-4"
  >
    <form>
      <h1>
        Únete a nuestra comunidad!
      </h1>
      <div
        className="form-group"
      >
        <label
          className="control-label"
        >
          Email
        </label>
        <input
          className="form-control"
          name="email"
          placeholder="Introduzca su email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label"
        >
          Usuario
        </label>
        <input
          className="form-control"
          name="login"
          placeholder="Introduzca su nombre de usuario"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label"
        >
          Password
        </label>
        <input
          className="form-control"
          name="password"
          placeholder="Introduzca su contraseña"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group"
      >
        <a
          className="btn btn-primary btn-lg"
          onClick={[Function]}
        >
          Regístrate
        </a>
      </div>
    </form>
  </div>
</div>
`;
